apps/aecore/src/aec_sync.erl

```diff
@@ -109,9 +109,34 @@ handle_worker(Task, Action) ->
 %% We sync with several nodes at the same time and use as strategy
 %% to pick a random hash from the hashes in the pool.
 
--record(sync_task, {id, chain, pool = [], agreed,
-                    adding = [], pending = [],
-                    workers = [] :: [{PeerId::binary(), pid()}]}).
+-type chain_id() :: reference().
+-record(chain_block, { hash :: aec_blocks:block_header_hash()
```

Comment:
Wouldn't `chain_key_block` be more appropriate?!

Reply:
One such item is build taking `aec_headers:prev_hash(Header)` - not the prev key hash - so it seems no to me. 🤔

Reply:
😱 That sounds like a bug! I'm not within easy reach of a computer so I can't check properly - but I thought sync would work on key-blocks (headers) 🤔

Reply:
[Tracked](https://www.pivotaltracker.com/story/show/167589944).